fe/fe-core/src/main/java/com/starrocks/meta/SqlDigestBlackList.java

```diff
@@ -0,0 +1,128 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.meta;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.staros.util.LockCloseable;
+import com.starrocks.common.AnalysisException;
+import com.starrocks.common.ErrorCode;
+import com.starrocks.common.ErrorReport;
+import com.starrocks.metric.MetricRepo;
+import com.starrocks.persist.DeleteSqlDigestBlackLists;
+import com.starrocks.persist.ImageWriter;
+import com.starrocks.persist.SqlDigestBlackListPersistInfo;
+import com.starrocks.persist.metablock.SRMetaBlockEOFException;
+import com.starrocks.persist.metablock.SRMetaBlockException;
+import com.starrocks.persist.metablock.SRMetaBlockID;
+import com.starrocks.persist.metablock.SRMetaBlockReader;
+import com.starrocks.persist.metablock.SRMetaBlockWriter;
+import com.starrocks.server.GlobalStateMgr;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+public class SqlDigestBlackList {
+    private static final Logger LOG = LogManager.getLogger(SqlDigestBlackList.class);
+
+    private final ReadWriteLock rwLock = new ReentrantReadWriteLock();
+
+    private final Set<String> sqlDigestBlackList = ConcurrentHashMap.newKeySet();
+
+    public Set<String> getDigests() {
+        try (LockCloseable ignored = new LockCloseable(rwLock.readLock())) {
+            return new TreeSet<>(sqlDigestBlackList);
+        }
+    }
+
+    public void verifying(String digest) throws AnalysisException {
+        try (LockCloseable ignored = new LockCloseable(rwLock.readLock())) {
+            if (sqlDigestBlackList.contains(digest)) {
+                MetricRepo.COUNTER_SQL_BLOCK_HIT_COUNT.increase(1L);
+                ErrorReport.reportAnalysisException(
+                        ErrorCode.ERR_SQL_IN_BLACKLIST_ERROR.formatErrorMsg() + ". Digest: %s",
+                        ErrorCode.ERR_SQL_IN_BLACKLIST_ERROR,
+                        digest);
+            }
+        }
+    }
+
+    public void addDigest(String digest) {
+        if (digest == null || digest.isEmpty()) {
+            return;
+        }
+        try (LockCloseable ignored = new LockCloseable(rwLock.writeLock())) {
+            GlobalStateMgr.getCurrentState().getEditLog()
+                    .logAddSqlDigestBlackList(new SqlDigestBlackListPersistInfo(digest),
+                            wal -> sqlDigestBlackList.add(digest));
+        }
+    }
+
+    public void put(String digest) {
```

Comment:
What's the diff for use between `put` and `addDigest`?

Reply:
addDigest/deleteDigests are with logging to editlog, and called by stmeExecutor.
put/deleteAll are without logging to editlog, and called when load from journal.